From 8faac273d3fa2394fbc686be52476bd229754338 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Mon, 22 Mar 2010 14:20:56 -0700
Subject: [PATCH 11/14] XKB: Fix garbage initialization

XkbEnableDisableControls set extra garbage bits on the xkbControlsNotify
changedControls mask because it was uninitialized on the stack.

Found by clang

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
(cherry picked from commit e081ef1d26c03a38da58f1eacf21ebe8fff5a7d9)
---
 xkb/xkbUtils.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git xorg-server-1.7.6/xkb/xkbUtils.c xorg-server-1.7.6/xkb/xkbUtils.c
index 75e243c..30ec438 100644
--- xorg-server-1.7.6/xkb/xkbUtils.c
+++ xorg-server-1.7.6/xkb/xkbUtils.c
@@ -753,7 +753,7 @@ XkbSrvLedInfoPtr	sli;
     if (cause!=NULL) {
 	xkbControlsNotify cn;
 	cn.numGroups= ctrls->num_groups;
-	cn.changedControls|= XkbControlsEnabledMask;
+	cn.changedControls= XkbControlsEnabledMask;
 	cn.enabledControls= ctrls->enabled_ctrls;
 	cn.enabledControlChanges= (ctrls->enabled_ctrls^old);
 	cn.keycode= cause->kc;
-- 
1.7.0.2

